src/test/java/com/faforever/server/player/PlayerServiceTest.java

```diff
@@ -35,29 +35,44 @@
   @Before
   public void setUp() throws Exception {
     player = (Player) new Player().setId(1);
+    player.setLogin("JUnit");
     instance = new PlayerService(clientService);
   }
 
   @Test
   public void onClientDisconnectRemovesPlayerAndUnsetsGameAndRemovesGameIfLastPlayer() throws Exception {
-    User user = new User();
-    user.setPassword("pw");
-    user.setLogin("JUnit");
-    user.setCountry("CH");
-    user.setPlayer(player);
-
-    player.setClientConnection(new ClientConnection("1", Protocol.LEGACY_UTF_16, mock(InetAddress.class)));
+    FafUserDetails fafUserDetails = mockFafUserDetails();
 
-    FafUserDetails fafUserDetails = new FafUserDetails(user);
     instance.onAuthenticationSuccess(new AuthenticationSuccessEvent(new TestingAuthenticationToken(fafUserDetails, "pw")));
     assertThat(instance.getPlayer(player.getId()).isPresent(), is(true));
 
     InetAddress inetAddress = mock(InetAddress.class);
     ClientConnection clientConnection = new ClientConnection("1", Protocol.LEGACY_UTF_16, inetAddress)
-      .setUserDetails(new FafUserDetails(user));
+      .setUserDetails(new FafUserDetails(fafUserDetails.getUser()));
 
     instance.onClientDisconnect(new ClientDisconnectedEvent(this, clientConnection));
 
     assertThat(instance.getPlayer(player.getId()).isPresent(), is(false));
   }
+
+  @Test
+  public void isPlayerOnline() {
+    FafUserDetails fafUserDetails = mockFafUserDetails();
+
+    assertThat(instance.isPlayerOnline(fafUserDetails.getUser().getPlayer().getLogin()), is(false));
+    instance.onAuthenticationSuccess(new AuthenticationSuccessEvent(new TestingAuthenticationToken(fafUserDetails, "pw")));
+    assertThat(instance.isPlayerOnline(fafUserDetails.getUser().getPlayer().getLogin()), is(true));
+  }
+
+  private FafUserDetails mockFafUserDetails() {
```

Comment:
Since it's technically not a mock, I recommend to call it `createFafUserDetails()`;